provisioning/mypy.ini

```diff
@@ -0,0 +1,8 @@
+[mypy]
+python_version = 3.6
```

Comment:
I think we also need `check-untyped-defs` here (see https://github.com/Seagate/cortx-hare/pull/1411)

Please consider https://github.com/Seagate/cortx-hare/blob/main/hax/mypy.ini as a basic reference.

Reply:
Added check-untyped-defs.